### PR TITLE
Bad node cleanup and covenant logic

### PR DESCRIPTION
**Pull request by Chronial**
* Move constants to `Config.vb`
* Automatic disconnecting of useless nodes
* Smarter DSCM-Net node picking logic, use covenant to choose better nodes
* Color nodes in active node list based on usefulness

This is mostly waiting for a stable release, which should definitely happen before this is merged. Also some review would probably be a good idea:
- [ ] Test ring detection (in untested)
- [ ] Review SL range calculations
- [ ] Review IRC Node selection logic
- [ ] Review disconnection logic

### CODE REVIEW

**Comment by Wulf2k** [comment]
I'll try get the stable out tonight, and then look at this more on the
weekend.

**Comment by Wulf2k** [comment]
Geez, I let that go this long without merging?

Sorry about that.

File: DaS-PC-MPChan/MainWindow.vb [hunk 1]
```diff
@@ -365,22 +366,177 @@ Public Class MainWindow
...
+            If blackSet.Contains(node.SteamId) Then Continue For
+            candidates.Add(node)
+        Next
+
+        If candidates.Count = 0 Then Return Nothing
+
+        Dim self = dsProcess.SelfNode
+
+        'These read out dsProcess memory, so don't calculate them for every node
+        Dim anorLondoInvading = self.Covenant = Covenant.DarkmoonBlade AndAlso dsProcess.HasDarkmoonRingEquiped
```
Comment (Wulf2k) [inline DaS-PC-MPChan/MainWindow.vb]: Equiped should be Equipped     (pp)

File: DaS-PC-MPChan/MainWindow.vb [hunk 2]
```diff
@@ -365,22 +366,177 @@ Public Class MainWindow
...
+                                     .ThenByDescending(Function(other) other.canBeRedSignSummoned(self)) _
+                                     .ThenByDescending(Function(other) other.canRedEyeInvade(self)) _
+                                     .ThenByDescending(Function(other) self.canRedEyeInvade(other)) _
+                                     .ThenByDescending(Function(other) other.canBeSummoned(self)) _
+                                     .ThenByDescending(Function(other) self.canBeSummoned(other))
+                         End Function
+
+        Dim sameMpZone = Function(other) other.MPZone = self.MPZone
+
+        If self.Covenant = Covenant.Darkwraith Then
```
Comment (Wulf2k) [inline DaS-PC-MPChan/MainWindow.vb]: Is this functionally a "Select Case" structure or do the ElseIfs add something to it?

Comment (Chronial) [inline DaS-PC-MPChan/MainWindow.vb]: No, this is just a case select.

**Pull request merged**